### PR TITLE
Ivf c example

### PR DESCRIPTION
**Pull request by abner-ma**
Add examples of ivf-flat and ivf-pq in C language

### CODE REVIEW

File: examples/c/src/ivf_pq_c_example.c [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    cuvsIvfPqSearchParams_t search_params;
+    cuvsIvfPqSearchParamsCreate(&search_params);
+    search_params->n_probes = 50;
+    search_params->internal_distance_dtype = CUDA_R_16F;
+    search_params->lut_dtype = CUDA_R_16F;
+
+    // Search the `index` built using `cuvsIvfPqBuild`
+    cuvsError_t search_status = cuvsIvfPqSearch(*res, search_params, index,
+     queries_tensor, &neighbors_tensor, &distances_tensor);
+    if (build_status != CUVS_SUCCESS) {
```
Comment (benfred) [inline examples/c/src/ivf_pq_c_example.c:77]: You probably meant to check the search_status here ? (build_status was checked above and is guaranteed to be success)

```suggestion
    if (search_status != CUVS_SUCCESS) {
```

Comment (benfred) [reply examples/c/src/ivf_pq_c_example.c:77]: Also - should we just abort the example if the search failed ?

File: examples/c/src/ivf_flat_c_example.c [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    
+    // Create default search params
+    cuvsIvfFlatSearchParams_t search_params;
+    cuvsIvfFlatSearchParamsCreate(&search_params);
+    search_params->n_probes = 10;
+
+    // Search the `index` built using `ivfFlatBuild`
+    cuvsError_t search_status = cuvsIvfFlatSearch(*res, search_params, index,
+     queries_tensor, &neighbors_tensor, &distances_tensor);
+    if (build_status != CUVS_SUCCESS) {
```
Comment (benfred) [inline examples/c/src/ivf_flat_c_example.c:170]:
```suggestion
    if (search_status != CUVS_SUCCESS) {
```

**Review by benfred — COMMENTED** [review]
thanks for this example! The changes mostly look good to me, aside from a couple minor issues:

File: examples/c/src/ivf_pq_c_example.c (hunk 1 above)
Comment (abner-ma) [reply examples/c/src/ivf_pq_c_example.c]: Yes. So I modified and add  'exit(-1);'

**Review by benfred — APPROVED** [review]

**Pull request merged**